### PR TITLE
feat: :sparkles: adds iam consent validation plugin

### PR DESCRIPTION
**Pull request by dsoffiantini**
## Description

Adds a new validator for IAM consent

## Motivation and Context

Validates when consent is active, messaging has been configured to use it properly

## How Has This Been Tested?
Unit tests added for all possible results

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: plugins/aep-iam-consent/index.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  ) as SharedStateVersions[];
+
+  const isConsentInstalled = versionEvents.some(
+    versions.getExtensionsKey('"com.adobe.edge.consent"')
+  );
+
+  if (!isConsentInstalled) {
+    return {
+      message:
+        'Consent has been configured inside launch, but the extension has not been registered in your application',
```
Comment (dgrossen) [inline plugins/aep-iam-consent/index.ts:55]: Data Collection instead of launch

Comment (dgrossen) [reply plugins/aep-iam-consent/index.ts:55]: Could also add https://github.com/AdobeDocs/aep-mobile-sdkdocs/blob/main/src/pages/documentation/getting-started/tabs/get-the-sdk.md as a `link`

Comment (dsoffiantini) [reply plugins/aep-iam-consent/index.ts]: @dgrossen updated!

File: plugins/aep-iam-consent/index.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      result: 'not matched',
+      links: [
+        {
+          type: 'doc',
+          url: 'https://developer.adobe.com/client-sdks/documentation/consent-for-edge-network/'
+        }
+      ]
+    };
+  }
+
```
Comment (amadt) [inline plugins/aep-iam-consent/index.ts:70]: What about if neither are true?

**Review by dgrossen — APPROVED** [review]

**Review by amadt — APPROVED** [review]

**Review by dgrossen — APPROVED** [review]

**Pull request merged**